tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -206,9 +211,13 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
                              .WithKeys(ConfigurationKeys.BufferSize)
                              .AsInt32(defaultValue: 1024 * 1024 * 10); // 10MB
 
+            // If Lambda/GCP we don't wanat to have a flush interval. The serverless integration
+            // manually calls flush and waits for the result before ending execution.
+            // This can artificially increase the execution time of functions
+            var defaultTraceBatchInterval = IsRunningInLambda || IsRunningInGCPFunctions ? 0 : 100;
```

Comment:
why don't you include `AzureFunctionsConsumptionsPlan`? I know it's not used yet (same for GCPFunctions btw), but It's what you've done for telemetry so I assume it could be relevant here as well no?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/4946/commits/9e9644ed3d89be931f39f164d926989c2f0fec87